openbas-api/src/test/java/io/openbas/rest/InjectApiTest.java

```diff
@@ -265,6 +267,71 @@ void deleteInjectsForScenarioTest() throws Exception {
 
   // -- EXERCISES --
 
+
+  @DisplayName("Add an inject for simulation")
+  @Test
+  @Order(9)
+  @WithMockPlannerUser
+  void addInjectForSimulationTest() throws Exception {
+    // -- PREPARE --
+    InjectInput input = new InjectInput();
+    input.setTitle("Test inject");
+    input.setInjectorContract(EMAIL_DEFAULT);
+    input.setDependsDuration(0L);
+
+    // -- EXECUTE --
+    String response = mvc
+        .perform(post(EXERCISE_URI + "/" + EXERCISE.getId() + "/injects")
+            .content(asJsonString(input))
+            .contentType(MediaType.APPLICATION_JSON)
+            .accept(MediaType.APPLICATION_JSON))
+        .andExpect(status().is2xxSuccessful())
+        .andReturn()
+        .getResponse()
+        .getContentAsString();
+
+    // -- ASSERT --
+    assertNotNull(response);
+    EXERCISE_INJECT_ID = JsonPath.read(response, "$.inject_id");
+    response = mvc
+        .perform(get(EXERCISE_URI + "/" + EXERCISE.getId())
+            .accept(MediaType.APPLICATION_JSON))
+        .andExpect(status().is2xxSuccessful())
+        .andReturn()
+        .getResponse()
+        .getContentAsString();
+    assertEquals(EXERCISE_INJECT_ID, JsonPath.read(response, "$.exercise_injects[0]"));
+  }
+
+  @DisplayName("Update inject for simulation")
+  @Test
+  @Order(10)
+  @WithMockPlannerUser
+  void updateInjectForSimulationTest() throws Exception {
+    // -- PREPARE --
+    Inject inject = injectRepository.findById(EXERCISE_INJECT_ID).orElseThrow();
+    InjectInput input = new InjectInput();
+    String injectTitle = "A new title";
+    input.setTitle(injectTitle);
+    input.setInjectorContract(inject.getInjectorContract().map(InjectorContract::getId).orElse(null));
```

Comment:
Why are you setting the injector contract, if you don't check it in the assert part ?